### PR TITLE
add PSC support

### PR DESCRIPTION
**Pull request by bcantrill**

### CODE REVIEW

File: app/psc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+[features]
+itm = ["panic-itm", "kern/klog-itm"]
+semihosting = ["panic-semihosting", "kern/klog-semihosting"]
+
+[dependencies]
+cortex-m = { version = "0.7", features = ["inline-asm"] }
+cortex-m-rt = "0.6.12"
+panic-itm = { version = "0.4.1", optional = true }
+panic-halt = { version = "0.2.0", optional = true }
```
Comment (cbiffle) [inline app/psc/Cargo.toml:15]: There's no actual way to turn this on by feature, you can remove it.

File: app/psc/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+panic-semihosting = { version = "0.5.3", optional = true }
+cfg-if = "0.1.10"
```
Comment (cbiffle) [inline app/psc/Cargo.toml:17]: You don't use this, you can remove it.

**Review by cbiffle — APPROVED** [review]

File: app/psc/app.toml [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+[[config.i2c.controllers]]
+controller = 2
+
+#
+# SP_I2C_LOCAL_SDA
+# SP_I2C_LOCAL_SCL
+#
+[config.i2c.controllers.ports.F]
+name = "local"
+description = "M.2 bus"
```
Comment (rmustacc) [inline app/psc/app.toml:187]: I don't think there's an M.2 bus here.

**Review by rmustacc — COMMENTED** [review]
While trying to go through the app.toml, noticed one small thing there on i2c, though I haven't had a chance to go through the full schematic and supporting docs on this yet that's required for the rest of the app.toml. On the other hand if it works that probably at least is a fairly good sign.

File: app/psc/Cargo.toml (hunk 1 above)
Comment (bcantrill) [reply app/psc/Cargo.toml]: Fixed

File: app/psc/Cargo.toml (hunk 2 above)
Comment (bcantrill) [reply app/psc/Cargo.toml]: Fixed

File: app/psc/app.toml (hunk 3 above)
Comment (bcantrill) [reply app/psc/app.toml:187]: Ha -- yes, thank you! Fixed the description.

**Pull request merged**